The error during the configure stage should be something like:
```
In file included from ./waftools/fragments/cocoa.m:3:
In file included from /nix/store/jj45q9jrzzrjqy00f53xdj87qncvc0sk-apple-framework-Cocoa/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
In file included from /nix/store/936h4qrq947hcdpcz7xl9dbhvcj0m01v-apple-framework-Foundation/Library/Frameworks/Foundation.framework/Headers/Foundation.h:141:
In file included from /nix/store/936h4qrq947hcdpcz7xl9dbhvcj0m01v-apple-framework-Foundation/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:7:
In file included from /nix/store/c0i2nq4vwcfnnf9a85nrh5xrby7mnc6c-apple-framework-ApplicationServices/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:39:
In file included from /nix/store/8ir16jcsxhr8fjfsm9ygvsqxlbm1znis-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CoreText.h:27:
In file included from /nix/store/8ir16jcsxhr8fjfsm9ygvsqxlbm1znis-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CTFramesetter.h:22:
In file included from /nix/store/8ir16jcsxhr8fjfsm9ygvsqxlbm1znis-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CTTypesetter.h:21:
/nix/store/8ir16jcsxhr8fjfsm9ygvsqxlbm1znis-apple-framework-CoreText/Library/Frameworks/CoreText.framework/Headers/CTLine.h:22:10: fatal error: 'CoreFoundation/CFAttributedString.h' file not found
#include <CoreFoundation/CFAttributedString.h>
         ^
66 warnings and 1 error generated.
```

--- a/pkgs/applications/video/mpv/default.nix
+++ b/pkgs/applications/video/mpv/default.nix
@@ -1,7 +1,7 @@
 { stdenv, fetchurl, fetchFromGitHub, makeWrapper
 , docutils, perl, pkgconfig, python3, which, ffmpeg
 , freefont_ttf, freetype, libass, libpthreadstubs
-, lua, lua5_sockets, libuchardet, rubberband
+, lua, lua5_sockets, libuchardet, libiconv ? null, darwin
 
 , x11Support ? true,
     mesa       ? null,
@@ -13,18 +13,19 @@
     wayland      ? null,
     libxkbcommon ? null
 
+, rubberbandSupport  ? !stdenv.isDarwin, rubberband ? null
 , xineramaSupport    ? true,  libXinerama   ? null
 , xvSupport          ? true,  libXv         ? null
 , sdl2Support        ? true,  SDL2          ? null
-, alsaSupport        ? true,  alsaLib       ? null
+, alsaSupport        ? !stdenv.isDarwin,  alsaLib       ? null
 , screenSaverSupport ? true,  libXScrnSaver ? null
 , vdpauSupport       ? true,  libvdpau      ? null
-, dvdreadSupport     ? true,  libdvdread    ? null
-, dvdnavSupport      ? true,  libdvdnav     ? null
+, dvdreadSupport     ? !stdenv.isDarwin,  libdvdread    ? null
+, dvdnavSupport      ? dvdreadSupport,  libdvdnav     ? null
 , bluraySupport      ? true,  libbluray     ? null
 , speexSupport       ? true,  speex         ? null
 , theoraSupport      ? true,  libtheora     ? null
-, pulseSupport       ? true,  libpulseaudio ? null
+, pulseSupport       ? !stdenv.isDarwin,  libpulseaudio ? null
 , bs2bSupport        ? true,  libbs2b       ? null
 , cacaSupport        ? true,  libcaca       ? null
 , libpngSupport      ? true,  libpng        ? null
@@ -39,11 +40,12 @@
 
 with stdenv.lib;
 
-let 
+let
   available = x: x != null;
 in
 assert x11Support         -> all available [mesa libX11 libXext libXxf86vm];
 assert waylandSupport     -> all available [wayland libxkbcommon];
+assert rubberbandSupport  -> available rubberband;
 assert xineramaSupport    -> x11Support && available libXinerama;
 assert xvSupport          -> x11Support && available libXv;
 assert sdl2Support        -> available SDL2;
@@ -109,7 +111,7 @@ in stdenv.mkDerivation rec {
 
   buildInputs = [
     ffmpeg freetype libass libpthreadstubs
-    lua lua5_sockets libuchardet rubberband
+    lua lua5_sockets libuchardet
   ] ++ optional alsaSupport        alsaLib
     ++ optional xvSupport          libXv
     ++ optional theoraSupport      libtheora
@@ -118,6 +120,10 @@ in stdenv.mkDerivation rec {
     ++ optional bluraySupport      libbluray
     ++ optional jackaudioSupport   libjack2
     ++ optional pulseSupport       libpulseaudio
+    ++ optional stdenv.isDarwin    libiconv
+    ++ optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+       Cocoa CoreAudio ])
+    ++ optional rubberbandSupport  rubberband
     ++ optional screenSaverSupport libXScrnSaver
     ++ optional vdpauSupport       libvdpau
     ++ optional speexSupport       speex
@@ -158,7 +164,7 @@ in stdenv.mkDerivation rec {
     homepage = http://mpv.io;
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ AndersonTorres fuuzetsu ];
-    platforms = platforms.linux;
+    platforms = platforms.darwin ++ platforms.linux;
 
     longDescription = ''
       mpv is a free and open-source general-purpose video player,

--- a/pkgs/top-level/lua-packages.nix
+++ b/pkgs/top-level/lua-packages.nix
@@ -192,6 +192,11 @@ let
     patchPhase = ''
       sed -e "s,^LUAPREFIX_linux.*,LUAPREFIX_linux=$out," \
           -i src/makefile
+    '' + stdenv.lib.optionalString stdenv.isDarwin ''
+      export PLAT=macosx
+      export LUAPREFIX_macosx=$out
+      substituteInPlace src/Makefile --replace gcc cc \
+        --replace 10.3 10.5
     '';
 
     meta = {

